docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,200 @@
+
+## Decision
+
+We think that adding Rosetta API support to the Cosmos SDK will bring value to all the developers and Cosmos SDK based chains in the ecosystem.
+
+
+## Architecture
+
+The driving principles of the proposed design follow:
+
+1. Developer-friendliness: it must be as riskless and painless as possible for client applications developers to extend network configurations to expose Rosetta API-compliant services.
+2. Long term support: developers build applications on the various stable branches of Cosmos SDK. This proposal aims to provide support for all the release series supported by the Cosmos SDK team.
+3. Cost-efficiency: backporting features from `master` to the various stable branches of Cosmos SDK is a cost that needs to be reduced.
+
+### Service
+
+`Service` is a simple `struct` that is started and listens to the port specified in the options. This is meant to be used across all the Cosmos SDK versions that are actively supported.
+
+The constructor follows:
+
+`func New(options Options, network Network) (*Service, error)`
+
+It accepts an `Options` `struct` that holds service configuration values, such as the port the service would be listening to:
+
+```
+type Options struct {
+	Port uint32
+}
+```
+
+The `Network` type holds network-specific properties (i.e. configuration values) and adapters. Ready-to-use Cosmos SDK release series-specific base `Network` concrete types will be made available. Applications can embed such types in their own custom types.
+
+```
+type Network struct {
+	Properties rosetta.NetworkProperties
+	Adapter    rosetta.Adapter
+}
+````
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service` to run:
+
+```
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	AddrPrefix          string
+	SupportedOperations []string
+}
+```
+
+Rosetta API services use Blockchain and Network as identifiers, e.g. the developers of gaia, the application that powers the Cosmos Hub, may want to set those to Cosmos Hub and cosmos-hub-3 respectively.
+
+`AddrPrefix` contains the network-specific address prefix. Cosmos SDK base implementations will default to `cosmos`, client applications are instructed that this should be changed according to their network configuration.
+
+`SupportedOperations` contains the transaction types that are supported by the library. At the present time, only `Transfer` is supported.
+
+`Network` holds an `Adapter` reference too. Adapter implementations may vary across different Cosmos SDK release series:
+
+```
+type Adapter interface {
+	DataAPI
+	ConstructionAPI
+}
+
+type DataAPI interface {
+	server.NetworkAPIServicer
+	server.AccountAPIServicer
+	server.MempoolAPIServicer
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+
+````
+
+### Cosmos SDK Integration
```

Comment:
I think this section is trying to explain how to use this new functionality. If so, should explain this at the start of this section.